pkg/controller/tidbcluster/tidb_cluster_control.go

```diff
@@ -174,6 +178,23 @@ func (c *defaultTidbClusterControl) updateTidbCluster(tc *v1alpha1.TidbCluster)
 		return err
 	}
 
+	if features.DefaultFeatureGate.Enabled(features.VolumeReplacing) {
+		if err := c.pvcReplacer.UpdateStatus(tc); err != nil {
+			metrics.ClusterUpdateErrors.WithLabelValues(ns, tcName, "pvc_replacer_updatestatus").Inc()
+			return err
+		}
+	}
+
+	// syncing the labels from Pod to PVC and PV, these labels include:
+	//   - label.StoreIDLabelKey
+	//   - label.MemberIDLabelKey
+	//   - label.NamespaceLabelKey
+	// Also updates the TiKV pod annotation for label.AnnTiKVNoActiveStoreSince if store is missing.
+	if err := c.metaManager.Sync(tc); err != nil {
```

Comment:
> Move MetaManager.Sync() in tidb_cluster_control.go which sets tikv store labels to run before component member manager Sync's so that label is updated, even if member managers error.

Could it cause a deadlock if meta sync failed as the member manager can't sync some necessary operations?

Reply:
It's a fair concern, I've gone through the metamanger Sync to see what errors it could possibly throw, and they are usually errors from kubernetes operations which should be safe to retry without blocking on a member sync / deadlock.
https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/meta/meta_manager.go#L39
https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/pod_control.go#L107
https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/pv_control.go#L121
https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/pvc_control.go#L144

The only concerning portion that stood out to me is that there is a pdclient call:
https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/pod_control.go#L126C34-L126C34

Just to be safe I've moved this after pd member manager, Other than that I can't see any errors that would depend on member manager to resolve. The only new dependency is that this runs before tikv member manager.

Reply:
#5140 will change the sync order, and TiCDC may need to be synced before PD

Reply:
Ah ok, I've changed the code to make this call twice now. Once before all the member managers but without blocking on error. And the second after all the member managers which blocks on error (existing old location). Should be safe to do this twice since it's idempotent, and the first one will now ignore error to avoid deadlock.  Do you think this is okay?